chat.js

```diff
@@ -727,10 +729,12 @@ class CommandContext {
 				return false;
 			}
 
-			// If the corresponding config option is set, non-AC users cannot send links, except to staff.
-			if (Config.restrictLinks && !user.autoconfirmed && message.match(Chat.linkRegex)) {
-				if (!(targetUser && targetUser.can('lock'))) {
-					this.errorReply("Your account must be autoconfirmed to send links to other users, except for global staff.");
+			// If the corresponding Config option is set, non-AC users cannot send links, except to staff.
+			const links = message.match(Chat.linkRegex);
+			if (Config.restrictLinks && !user.autoconfirmed && links) {
+				const whitelisted = !links.some(link => !AC_LINK_WHITELIST.includes(/[^/]+(?=$|\/(?!\/))/.exec(link)[0]));
```

Comment:
I would pull `/[^/]+(?=$|\/(?!\/))/.exec(link)[0]` out into its own variable; it's hard to read smushed together.

Regexes in general are kind of hard to read tbh.

...honestly, I have a hard time believing it's what you're looking for.

Reply:
`/[^/]+(?=$|\/(?!\/))/.exec('http://example.com//pokemonshowdown.com/this-is-definitely-not-on-the-whitelist')[0]`
`"pokemonshowdown.com"`

I was right! I _thought_ that regex looked fishy. Backreference magic is basically never the right way to do things. Instead of assuming that URLs follow rules, you need to understand which part of a URL is trustworthy and which part is untrustworthy, and construct your regex based on _that_ knowledge.

Reply:
`/^(?:http\:\/\/|https:\/\/)?(?:[^\/]*\.)?([^\/]*\.[^\/]*)\.?($|\/|:)/`

Here's my regex.